compiler/src/iree/compiler/DispatchCreation/CollapseDimensions.cpp

```diff
@@ -668,6 +668,7 @@ hoistTensorReshapesOutOfDispatchRegion(
       ValueRange resultDims = dispatchOp.getResultDynamicDims(resultIndex);
       newDynamicDims.append(resultDims.begin(), resultDims.end());
       newYieldVals.push_back(yieldedValue);
+      newDynamicDims.assign(dynamicDimsList.begin(), dynamicDimsList.end());
```

Comment:
If we assign it here, we do not need to append the resultDims in l.669?

Reply:
Actually I missed that. Let me look at this again